### PR TITLE
Implement seed phrase fallback flow

### PR DESCRIPTION
**Pull request by Parveshdhull**
original PR https://github.com/status-im/status-mobile/pull/20766

Designs: https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3338-16100&node-type=CANVAS&t=yeQnRzRlEua9vV6V-0

depends on https://github.com/status-im/status-mobile/pull/21160 (PR merged)

status-go pr: https://github.com/status-im/status-go/pull/5794

### Leftover
image
As per design we also need to show profile name, but currently we don't have access to that. So only installation-id will be shown in more details bottom sheet.

### Testing
- Test happy path of syncing
- Test fallback using correct seed phrase
- Test fallback using different seed phrase

status: ready

### CODE REVIEW

File: src/status_im/contexts/profile/profiles/view.cljs
Comment (Parveshdhull) [inline src/status_im/contexts/profile/profiles/view.cljs]: Feel free to ignore this file. As its already reviewed in https://github.com/status-im/status-mobile/pull/21160

File: src/status_im/contexts/onboarding/syncing/progress/view.cljs
Comment (flexsurfer) [inline src/status_im/contexts/onboarding/syncing/progress/view.cljs:31]: why is only this event debounced?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/contexts/onboarding/syncing/progress/view.cljs
Comment (Parveshdhull) [reply src/status_im/contexts/onboarding/syncing/progress/view.cljs:31]: Thank you, @flexsurfer, for reviewing the PR and spotting this issue. ~The intention was to use setTimeout to add a delay between navigation, but debounce was used by mistake. (fixed now)~

Comment (Parveshdhull) [reply src/status_im/contexts/onboarding/syncing/progress/view.cljs:31]: We are using debounce-and-dispatch here to introduce a delay between navigating back to the `sync-or-recover-profile` screen and moving to the `sign-in-by-seed-phrase` screen.

**Review by qfrank — APPROVED** [review]

**Comment by qfrank** [comment]
***TESTING NOTE:***
**known issue:** display name might not get updated on new device if the testing way is turning off/on wifi.

**solution:**
if we use another way to test rather than turn off/on wifi, the fallback works well, the way is make 2 devices are not in the same LAN(e.g. one use wifi one use mobile data), but both devices should be able to access internet. 

I guess there're some issues relate to waku when use the way turn off/on wifi, we can create a separate issue then.

@status-im/mobile-qa

File: src/legacy/status_im/pairing/core.cljs
Comment (ilmotta) [inline src/legacy/status_im/pairing/core.cljs]: Can be just an event vector.

```clojure
:on-success [:sanitize-messages-and-process-response]
```

File: src/status_im/contexts/syncing/events.cljs
Comment (ilmotta) [inline src/status_im/contexts/syncing/events.cljs]: As you know, `rf/defn` is deprecated, and for new code it's the same effort rewriting it to use `reg-event-fx`. I remember a long time ago we agreed to avoid mixing in the same namespace `reg-event-fx` and `rf/defn`, but since that time, all CCs have been using `reg-event-fx`, so I think there's no need for us to grow usages of `rf/defn` anymore, unless in events mixed with `rf/merge`, but those are becoming more rare.

File: src/status_im/contexts/shell/activity_center/notification/syncing/view.cljs
Comment (ilmotta) [inline src/status_im/contexts/shell/activity_center/notification/syncing/view.cljs]: @Parveshdhull, notifications have a central place to transform data to idiomatic Clojure. It's error prone to mix different cases, so it's better if we stick with kebab case so that we are good Clojure citizens 😄 The function to rename `installationId` is `legacy.status-im.data-store.activities/<-rpc`.

Side note: the preferred way is to move these legacy data store files to their respective `context/` directories, as we do in `contexts/wallet/data_store.cljs` or `contexts/network/data_store.cljs`. Just mentioning since it's related, but I'm not suggesting a change for this PR :)

**Review by ilmotta — APPROVED** [review]
Great work on this feature @Parveshdhull and @qfrank!

**Comment by Parveshdhull** [comment]
Hi @Horupa-Olena, Thank you very much for taking the PR for testing.

Pushed a small commit. (leftover from Icaro's review). (Please use latest commit for testing)

Also PR is ready for testing but depends on https://github.com/status-im/status-mobile/pull/21160 for merging. Will rebase this PR once https://github.com/status-im/status-mobile/pull/21160 is merged.

**Comment by Parveshdhull** [comment]
> Also PR is ready for testing but depends on https://github.com/status-im/status-mobile/pull/21160 for merging. Will rebase this PR once https://github.com/status-im/status-mobile/pull/21160 is merged

PR rebased after https://github.com/status-im/status-mobile/pull/21160 is merged

**Comment by Horupa-Olena** [comment]
@Parveshdhull Hi!
Thank you for your work! Could you please rebase this PR again so that I can perform the final test on the actual branch after the weekend?

**Comment by qfrank** [comment]
branch `feat/enable_sync_fallback` on status-go side rebased cc @Parveshdhull

**Comment by Horupa-Olena** [comment]
@Parveshdhull Thanks for your PR!
I found some issues and have a few questions about the logic.

### Design Issues:
**ISSUE 1:** The "Oops, something went wrong" screen has differences in the buttons compared to the Figma design. Please check it against the Figma design.
Here's how it looks in the app:
photo_2024-09-10_18-40-01

**ISSUE 2:** When clicking the "Try again" button, I am returned to the screen with the Sync code (I tested using the Sync code, not QR). According to the design, the user should be returned to the first screen, "Sync or recover profile." Is this the expected behavior or a bug?
(next issues also in this video)

https://github.com/user-attachments/assets/c57a0d1a-5e2c-4a5b-847d-7f86e225a3c4


**ISSUE 3:** When clicking the "Enter seed phrase" button, the "Sync or recover profile" screen briefly appears before transitioning to the next step where the user enters the seed phrase.

**ISSUE 4:** After creating a password during onboarding, this screen is missing: https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3338-72168&node-type=frame&t=h2aHliWk1VIpyOk2-0. Instead, the default screen with the text "Generating keys..." appears.

**ISSUE 5:** The notification with the code in the Activity Center doesn’t relate to any tab, and after being read, it disappears from the Activity Center. Is this the expected behavior or a bug?

### Questions:

1. What is supposed to happen when the user clicks the "Pair and Sync" button? Currently, nothing happens when I click it. I have to manually go to Profile -> Syncing -> Paired devices to pair another device. Should this notification appear when sync worked correctly, and we didn't have to use this fallback?
photo_2024-09-10_18-39-59

2.If I unpair device A1 in the settings of device A2, should I do the same on device A2 for device A1, or should it automatically unpair on both devices?

3.  What information is supposed to be fetched after syncing? Was this feature included in this PR?

**Comment by Parveshdhull** [comment]
Thank you @Horupa-Olena for testing the PR and finding these issues.

Issues 1 and 2 should be fixed now.

Issue 3 is expected. We have to navigate back to "Sync or recovery flow" before navigating to "Recovery phrase screen"

Issue 4: Yes, we are using the same onboarding flow as usual, which is why this screen is missing. If we want this screen to be part of the seed phrase recovery flow, I can include it in a follow-up update. However, we probably don’t want to add it, as we are trying to simplify the onboarding process. (And I suppose we removed this screen from the old flow 🤔)

> Issue 5: The notification with the code in the Activity Center doesn’t relate to any tab, and after being read, it disappears from the Activity Center. Is this the expected behavior or a bug?

Yes even in the designs it's not part of any tab. We can improve this in the future if needed. It should disappear once the pairing is a success (As per meeting notes).

Question 1: Please can you confirm if both devices are getting marked paired after a few minutes or after the app restarts or they never do?

Question 2&3: I think already answered in meeting?

**Comment by Horupa-Olena** [comment]
@Parveshdhull, thanks for your fix. I will review it and recheck the moment in Question 1.

Regarding ISSUE 3 — to me, it looks like a bug that needs to be fixed because the user doesn't interact with the "Sync or recover profile" screen, and it shouldn't appear in this flow. After confirming the sync code, the screen where the user enters the seed phrase should appear. Nothing else.

> However, we probably don’t want to reintroduce this screen, as we're trying to simplify the onboarding process. (I believe we removed this screen from the old flow 🤔).

Can anyone else confirm this? I missed the part when the onboarding was simplified. Or maybe Alisher could update the design to ensure everything is consistent, so we don’t have questions later like, "Why is it one way in the design and different in the app?"

As for ISSUE 5, if it’s not too difficult, could you fix it now so that this notification is shown in the system tab, please? In my opinion, it's important that the user can see this notification and come back to pair devices if they accidentally close it the first time.

**Comment by Parveshdhull** [comment]
> Regarding ISSUE 3 — to me, it looks like a bug that needs to be fixed because the user doesn't interact with the "Sync or recover profile" screen, and it shouldn't appear in this flow. After confirming the sync code, the screen where the user enters the seed phrase should appear. Nothing else.

Please check the 00:05 mark in Issue 2. The "Sync or Recover Profile" screen was the first screen the user accessed. We need to return to this screen to select a different flow (Recovery Phrase). I added a 500 ms delay to ensure the navigation transitions look smooth; otherwise, they might appear disjointed due to overlapping transitions. However, the video from your device does not show any transitions. Could you please check if transitions are turned off in your device settings?

> Can anyone else confirm this? I missed the part when the onboarding was simplified. Or maybe Alisher could update the design to ensure everything is consistent, so we don’t have questions later like, "Why is it one way in the design and different in the app?"

I’m unsure if we’ve logged discussions somewhere, but you can refer to some related PRs about simplification: [PR #20912](https://github.com/status-im/status-mobile/pull/20912) and [PR #20503](https://github.com/status-im/status-mobile/pull/20503).
But I see now we are not introducing a new screen here, just changing the title and image from "Generating Keys" to "Preparing Status for you...". I will address this. Thank you

**Comment by Parveshdhull** [comment]
> As for ISSUE 5, if it’s not too difficult, could you fix it now so that this notification is shown in the system tab, please? In my opinion, it's important that the user can see this notification and come back to pair devices if they accidentally close it the first time.

The file [activity_center.go](https://github.com/status-im/status-go/blob/09ee23a3d9f7101639621c7c192a3f402909aea2/protocol/activity_center.go#L27) does not have a notification type system. However, even for the system tab, we are using a status-go call with an [arbitrary notification type](https://github.com/status-im/status-mobile/blob/2bf3e6ce1312c6faf4b714ab2165795574a3367b/src/status_im/contexts/shell/activity_center/notification_types.cljs#L27) of 66614.

Related comment: `TODO: Replace with correct enum values once status-go implements them.`

What do you think, @jrainville? Should we implement a client-only change to display the installation notification type in the system tab, or should we address this in status-go?

**Comment by Horupa-Olena** [comment]
@Parveshdhull 
Issue 1-3 is fixed. Thank you!

I'm not quite sure about issue 4. Are we doing it as per the design? Because the mentioned PRs remove the Welcome screen, and in our flow, it is not used.

**Comment by Parveshdhull** [comment]
Hi @xAlisher, As per [designs](https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3338-72168&node-type=frame&t=QlAlzvnq7tdiGCMn-0) we need to show `Preparing Status for you...` screen after `Enable biometrics?`. But as you can see in the below video, there is nothing to prepare at this stage and the app is ready to use.

Please can you check if this screen is leftover or if we need to show it somehow? Thank you

https://github.com/user-attachments/assets/b109015e-3e8b-4391-b887-39f44c08297f



cc @Horupa-Olena

**Comment by Horupa-Olena** [comment]
@Parveshdhull Please check the following issues. I tried to explain them, but they are complicated, so feel free to ask if something is unclear.

### ISSUE 6: Different visual representation of device states after syncing

Steps:
1. Device A1 syncs with Device A.
**Actual result:**  
**Device A1**: immediately shows Device A in the list with the 'pairing' state, but the name of Device A is not displayed
**Device A:** in the settings, Device A1 is shown with the 'Unpair' state and with its name

3. On Device A, click the 'Pair and Sync' button in the sync popup.
**Device A1**: the name of the paired device appears in the list.
**Device A:** nothing will visually change on the list ,but pairing and syncing have occurred (if you send messages in 1-1 chats, they will appear on both devices).

**Expected results:**
In my opinion, both devices should change each other's states to "Pair" and visually display this to the user in the settings simultaneously after clicking the 'Pair and Sync' button.

Video from device A1: 

https://github.com/user-attachments/assets/ce7fd793-af8a-4e2d-aa5f-be73d91027f2

Screen from device A before and after click on button:
photo_2024-09-12_12-40-22
photo_2024-09-12_12-40-14

**Comment by Horupa-Olena** [comment]
@Parveshdhull I think the next issue is related to the previous one.

 ### ISSUE 7: The required popup is not displayed on the device that is syncing with another device.

**Steps:**

1. Device A1 syncs with Device A.
2. On Device A, click the 'Pair and Sync' button in the sync popup.
3. Device A2 syncs with Device A.

**Expected result:** A popup [Sync your profile](https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3451-12759&node-type=frame&t=xSY9Obm5guQvD38k-0)  is displayed on Device A2.

**Actual result:** A popup [New device detected](https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3451-12926&node-type=frame&t=xSY9Obm5guQvD38k-0) is displayed on Device A2. If a try sync device A3 with device A - the same result.

Video from device A2: 

https://github.com/user-attachments/assets/51d0310e-7524-4856-9d9f-20ecc415cf8e

Screenshots from device A after sync: 
photo_2024-09-12_13-02-31
photo_2024-09-12_13-02-28

**Comment by Parveshdhull** [comment]
@Horupa-Olena Issue 4 should be fixed now.

@flexsurfer @ilmotta Added a commit for adding prepare-status-screen. Please let me know if anything needs to be changed. https://github.com/status-im/status-mobile/pull/21090/commits/69cb8d9a2f8b6397b8c6e0bc7c5118c0b687754d

(For illustrations compression, I will create an issue and address separately. But we need to find a better method for that)

**Comment by Parveshdhull** [comment]
Issue 6: Please can you confirm if it still show Pair button even after app restart?
Issue 7: I guess when you recover account multiple times then old activity center notifications gets fetched and shown. @jrainville @qfrank  ?

**Comment by Horupa-Olena** [comment]
@Parveshdhull
A small update on ISSUE 6: The state changes only after re-login. If this is expected behavior, I think we need to provide this information to the user or even automatically re-login the user after clicking the 'Pair and Sync' button.

**Comment by Parveshdhull** [comment]
> @Parveshdhull A small update on ISSUE 6: The state changes only after re-login. If this is expected behavior, I think we need to provide this information to the user or even automatically re-login the user after clicking the 'Pair and Sync' button.

No, this is not the expected behavior. It seems the UI didn’t update for some reason. However, it doesn’t appear to be a blocker for the actual syncing functionality. I suggest we address this separately in a follow-up. What do you think?

**Comment by Horupa-Olena** [comment]
@Parveshdhull 

### ISSUE 8: Incorrect Screen Displayed During Fallback Flow
**Steps:**

1. Initiate the fallback flow.
2. Enter the correct seed phrase for syncing two devices.
3. Click 'Continue' (expected to open the screen for creating a password).
4. Return to the previous screen.
5. Enter the correct seed phrase, but from a different status profile not used on this device.
6. Click 'Continue' (expected to open the screen for creating a new account).
7. Return to the previous screen.
8. Enter the correct seed phrase for syncing two devices.
9. Click 'Continue'.

**Actual result:** The screen for creating a new account is opened (this happens every time if the user triggers account creation in the fallback flow).

https://github.com/user-attachments/assets/b35bc745-e785-45eb-a91b-9f7619d8883d

**Expected result:** The screen for creating a password should be opened.

**Comment by Horupa-Olena** [comment]
> No, this is not the expected behavior. It seems the UI didn’t update for some reason. However, it doesn’t appear to be a blocker for the actual syncing functionality. I suggest we address this separately in a follow-up. What do you think?

@Parveshdhull Please check ISSUE 8 and finaly write which issues will be reported separately.

**Comment by Parveshdhull** [comment]
Thank you @Horupa-Olena  for finding all these issues. I think at this point a quick summary will be helpful.

Issues 1-4 fixed
- [ ] Issue 4 leftover - Compress the image used in the "Preparing Status for you" screen
- [ ] Issue 5: Add syncing notification to System Tab (status-go)
- [x] Issue 6: On the recovered device - The syncing screen only shows paired label after the restart
- [x] Issue 7: When you recover the account multiple times using seed phrase fallback then "Pair and sync" is also shown/fetched on the new recovered device (status-go)
- [x] Issue 8: In seed phrase fallback recovery, if you first enter the seed phrase for the wrong account and press 'Continue,' then go back and enter the seed phrase for the correct account, you will lose the syncing state..

I think mostly these are minor issues and edge cases. So all can be addressed separately. wdyt?

**Comment by Horupa-Olena** [comment]
@Parveshdhull Yes, everything is great.

> Issue 4 leftover - Compress the image used in the "Preparing Status for you" screen
 Issue 5: Add syncing notification to System Tab (status-go)
 Issue 6: On the recovered device - The syncing screen only shows paired label after the restart
 Issue 7: When you recover the account multiple times using seed phrase fallback then "Pair and sync" is also shown/fetched on the new recovered device (status-go)
 Issue 8: In seed phrase fallback recovery, if you first enter the seed phrase for the wrong account and press 'Continue,' then go back and enter the seed phrase for the correct account, you will lose the syncing state..

Can you clarify, are issues 6-8 on me, and will you create the rest? Or should I report all of them?

**Comment by jrainville** [comment]
> What do you think, @jrainville? Should we implement a client-only change to display the installation notification type in the system tab, or should we address this in status-go?

The new types were added in the status-go PR: https://github.com/status-im/status-go/pull/5794/files#diff-5ec7b4bf47b0aa5efc18062c5db35bfc699f9f20ceb7a4b2dca5038b98751f9dR43-R44

When you ask which notifications you want, you need to pass it the types you want. 
So I think maybe you just need to pass it the new types like we do here: https://github.com/status-im/status-desktop/pull/15820/files#diff-ecbacaf3a6fa981ad571ba3033a4403294f9dbb8a7dc45e6dafa0d1bbfd6a907R188
(we don't have "System", so I just added it to the list of "All", but you can add it to both)

IMO, some of that logic should be moved to status-go at some point, but for now we can use what's there.

**Comment by jrainville** [comment]
> Issue 7: I guess when you recover account multiple times then old activity center notifications gets fetched and shown. @jrainville @qfrank ?

Yeah it's hard to confirm from the screenshots, but indeed, if you sync multiple times in the row the same account, you will fetch from the store nodes the same requests to pair. If you looked at the installation ID inside, you could confirm if it's an old one (if you remembered the old ones :smile: )

**Comment by Parveshdhull** [comment]
> Can you clarify, are issues 6-8 on me, and will you create the rest? Or should I report all of them?

Please feel free to create yourself, or I can create them. Both works for me.

**Comment by jrainville** [comment]
> Issue 7: I guess when you recover account multiple times then old activity center notifications gets fetched and shown. @jrainville @qfrank ?

Maybe there is a way to fix on status-go. If the request to pair has the installation ID of the "target", then we can confirm if that is us. If it's not intended for us, we should just ignore it.

I can take a look later

**Comment by Horupa-Olena** [comment]
@Parveshdhull I've created reports for issues 6-8 from your list. Please create reports for issues 4-5, as I think you can add some technical details from your side more effectively.

Also, I found a small issue with the [new screen](https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3338-72168&node-type=frame&t=TxMrL7G1uwxZDT1s-0) for devices with a screen width of 1080. Since it's a very popular size for Android devices, I'd like to propose changing the text **Preparing Status for you...** to **Preparing app for you...** or **Preparing Status....** This will help maintain consistency in font size for titles across all screens.
@xAlisher What do you think??
Screenshot 1

**Comment by Horupa-Olena** [comment]
@Parveshdhull And I hope it last one issue:)

### ISSUE 9:Profile name not displayed on the login page

**Steps:**

1. Device A1 syncs with Device A.
2. Complete the entire fallback flow, but do not confirm pairing on Device A (Device A use Wi-fi, device A1 use mob data).
3. Go to the profile settings on Device A1.
4. Log out of the profile.

**Expected result:** The profile on Device A1 shows 3 random names. The same name should appear on the login page.

**Actual result:** The profile on Device A1 shows 3 random names. However, no name is displayed on the login page.
йййййй
ццццц

**Comment:** If pairing is confirmed on Device A, the profile name is pulled in, and everything works correctly. 

@Parveshdhull  @jrainville  Can you clarify what the expected result should be in this case? Should the 3 random  name be displayed, or should it show the name of the profile we are syncing???

**Comment by xAlisher** [comment]
> @Parveshdhull I've created reports for issues 6-8 from your list. Please create reports for issues 4-5, as I think you can add some technical details from your side more effectively.
> 
> Also, I found a small issue with the [new screen](https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3338-72168&node-type=frame&t=TxMrL7G1uwxZDT1s-0) for devices with a screen width of 1080. Since it's a very popular size for Android devices, I'd like to propose changing the text **Preparing Status for you...** to **Preparing app for you...** or **Preparing Status....** This will help maintain consistency in font size for titles across all screens. @xAlisher What do you think?? Screenshot 1

Great idea @Horupa-Olena, let's change "Status" to the "app"

**Comment by jrainville** [comment]
> @Parveshdhull @jrainville Can you clarify what the expected result should be in this case? Should the 3 random name be displayed, or should it show the name of the profile we are syncing???

I'm not sure, but it shouldn't have an empty login name.

In this case, you didn't fetch any info at all from the account, so it should either have the 3 words name as the profile name everywhere, or the pubkey, but not empty.

This is probably an issue in status-go, but it's unrelated to the pairing fallback. We have a similar issue here: https://github.com/status-im/status-desktop/issues/15776

IMO, it's an edge case, so not super high priority to fix.

**Comment by Horupa-Olena** [comment]
@qfrank @Parveshdhull Could you, please, resolve the conflicts and rebase PR?

**Comment by jrainville** [comment]
> > Issue 7: I guess when you recover account multiple times then old activity center notifications gets fetched and shown. @jrainville @qfrank ?
> 
> Maybe there is a way to fix on status-go. If the request to pair has the installation ID of the "target", then we can confirm if that is us. If it's not intended for us, we should just ignore it.
> 
> I can take a look later

@Parveshdhull @Horupa-Olena I checked the code and the message that we send from the new instance is very generic.

Basically, it sends a message to itself, so anyone using the same profile will receive it and know that there is a new installation using that profile.
Then, you can press the "Pair and Sync" on any of the devices that you trust and it will pair on that device. The other might need to accept also.

So there is nothing to do to "fix" it. It's intended behaviour. It's also relatively edge case. I doubt people will be deleting their data and re-pairing over and over like we do when testing this PR :smile:

**Comment by ilmotta** [comment]
> (For illustrations compression, I will create an issue and address separately. But we need to find a better method for that)

Agreed @Parveshdhull and thank you for that. We really need to establish a process to not include assets unless they are compressed according to a script and make target. This should be a mandatory step in every illustration added. The script should probably accept at least one argument, to customize the amount of compression because this varies significantly.

But I think we shouldn't merge these ~1.2MB illustrations in the repo as much as we can. This increases significantly an already large repository. The follow-up PR will introduce files with a different checksum, then we pay the price of the old files in the repo forever.

**Comment by Horupa-Olena** [comment]
> > > Issue 7: I guess when you recover account multiple times then old activity center notifications gets fetched and shown. @jrainville @qfrank ?
> > 
> > 
> > Maybe there is a way to fix on status-go. If the request to pair has the installation ID of the "target", then we can confirm if that is us. If it's not intended for us, we should just ignore it.
> > I can take a look later
> 
> @Parveshdhull @Horupa-Olena I checked the code and the message that we send from the new instance is very generic.
> 
> Basically, it sends a message to itself, so anyone using the same profile will receive it and know that there is a new installation using that profile. Then, you can press the "Pair and Sync" on any of the devices that you trust and it will pair on that device. The other might need to accept also.
> 
> So there is nothing to do to "fix" it. It's intended behaviour. It's also relatively edge case. I doubt people will be deleting their data and re-pairing over and over like we do when testing this PR 😄

@jrainville The situation here isn't that I was constantly deleting data and syncing the same device with another several times. These were all different devices. 

If we consider a real-life example, it could be something like this: a user has two devices ( ipad and iphone), and they sync them with each other through the fallback flow. But then the user changes their mobile phone to a new one and syncs this third device with the ipad. And here is where the issue arises: instead of the expected [notification without a button](https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3347-48049&node-type=frame&t=cuNqe4YU7NhKHAGL-0), they will see the [notification ](https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=3347-47006&node-type=frame&t=cuNqe4YU7NhKHAGL-0) that is supposed to appear only on ipad.

I'm not saying this needs to be fixed right now, but it's a case that should be addressed because it exists and needs to be fixed to make the flow clearer for the user.

**Comment by Horupa-Olena** [comment]
@jrainville 
My point is that this process should be as simple and clear for users as possible, so they don't end up in a situation where they can't figure it out and just go looking for a more convenient app.

Because even for me, it's hard sometime to understand this complicated process after reviewing the design and our call. And for an average user, it would seem like an impossible task.

**Comment by Parveshdhull** [comment]
> [@qfrank](https://github.com/qfrank) [@Parveshdhull](https://github.com/Parveshdhull) Could you, please, resolve the conflicts and rebase PR?

Done, both PRs are rebased.

> [@Parveshdhull](https://github.com/Parveshdhull) I've created reports for issues 6-8 from your list. Please create reports for issues 4-5, as I think you can add some technical details from your side more effectively.

Thank you very much for creating issues for 6-8. I will create issues for all leftovers once PR is merged. (These issues don't exist in develop yet)

**Comment by Horupa-Olena** [comment]
Hi @Parveshdhull !
Thanks for your help with the PR.
The new screen is fine, but I noticed that the buttons on several screens have lost their color.
At first, I thought it was a glitch, but the issue is consistently reproducible. Could you check if the recent design changes might have affected the buttons?

**Comment by Parveshdhull** [comment]
Hi, thank you for finding this issue. It's unlikely that it's related to the PR. But just to be sure are you facing this on Android? And on which flow, fresh account? Please can you confirm, that you tried another device?

**Comment by Horupa-Olena** [comment]
@Parveshdhull This reproduces on both platforms (Android, iOS) and only in the fallback flow, regardless of the account.
When I create an account, perform a successful sync, and restore, the buttons on these screens have color.

**Comment by Parveshdhull** [comment]
hi @Horupa-Olena, Thank you for finding this issue. It should be fixed now

By the way, we missed this earlier, but it is also present in the [Issue 2 video](https://github.com/status-im/status-mobile/pull/21090#issuecomment-2341375243) at 1:15.

**Comment by jrainville** [comment]
> @jrainville My point is that this process should be as simple and clear for users as possible, so they don't end up in a situation where they can't figure it out and just go looking for a more convenient app.
> 
> Because even for me, it's hard sometime to understand this complicated process after reviewing the design and our call. And for an average user, it would seem like an impossible task.

That's a good point. Maybe that's something we should bring up with designers to address soon after.

While the flows are currently correct, they are a bit confusing like you pointed out.

**Comment by Horupa-Olena** [comment]
@Parveshdhull Thank you for your fix. No issues from my side. PR is ready to be merged!

**Comment by Parveshdhull** [comment]
> @Parveshdhull Thank you for your fix. No issues from my side. PR is ready to be merged!

cc @qfrank

**Comment by qfrank** [comment]
> > @Parveshdhull Thank you for your fix. No issues from my side. PR is ready to be merged!
> 
> cc @qfrank

waiting for PR approval from @igor-sirotin

**Comment by qfrank** [comment]
> > @Parveshdhull Thank you for your fix. No issues from my side. PR is ready to be merged!
> 
> cc @qfrank

status-go changes merged @Parveshdhull

**Pull request merged**
